service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -150,5 +143,19 @@ private final Respondent getFirstRespondent() {
     }
 
     private final CaseManagementOrder caseManagementOrder;
+
     private final Others others;
+
+    public String buildFirstRespondentFullName() {
```

Comment:
I find this unnecessary and it does two things. I'd rather see composite based approach where that methods finds first respondent. Then `Party.buildFullName()` is called to format name on the outcome of first method.

Please split it.